search/views.py

```diff
@@ -29,3 +29,23 @@ def get_serializer_context(self):
             )
             context.update({"saved_companies_pk": saved_companies_pk})
         return context
+
+
+class AdvancedSearchView(ListAPIView):
+    queryset = Profile.objects.active_only().prefetch_related(
+        "regions", "categories", "activities"
+    )
+    serializer_class = CompanyAdvancedSerializers
+    filter_backends = [
+        filters.SearchFilter,
+        filters.OrderingFilter,
+    ]
+    search_fields = [
+        "name",
+        "official_name",
+        "common_info",
+        "service_info",
+        "product_info",
+    ]
```

Comment:
While the built-in functionality provides relatively wide set of features, in most cases (from my experience — almost always) we're forced to dive into custom implementations anyway. 

So I usually write (kind of) custom filters anyway.

This is NOT an issue, just sharing some facts about the approach I'm using personally.

Reply:
I agree with you, but we don't have enough requirements. Maybe later (after getting requirements from client) we will improve this functionality.